buildroot/share/PlatformIO/scripts/offset_and_rename.py

```diff
@@ -45,7 +45,8 @@
 	def encrypt(source, target, env):
 		marlin.encrypt_mks(source, target, env, board.get("build.encrypt"))
 
-	marlin.add_post_action(encrypt);
+	if board.get("build.encrypt") != "":
```

Comment:
Is that a possibility, that `board.encrypt` exists but is set empty?

Reply:
i had to, to bypass the lerdge bootloader protection

```
[env:LERDGEK_stlink]
platform            = ${lerdge_common.platform}
extends             = env:LERDGEK_usb_flash_drive
extra_scripts       = ${stm32_variant.extra_scripts}
build_flags         = ${env:LERDGEK.build_flags} ${stm_flash_drive.build_flags} -DU30
platform_packages   = ${stm_flash_drive.platform_packages}
upload_protocol     = stlink
monitor_speed       = 250000
board_build.encrypt =
board_build.offset  = 0x0
board_upload.offset_address = 0x08000000
```

Reply:
Wouldn't removing the line altogether do the same thing?

Reply:
its extended... so inherited its the only way i found, if i remove the lerdge .py its the mks/generic one which do it :/

Reply:
Ah, I see. Maybe `env:LERDGEK_usb_flash_drive` and `env:LERDGEK_stlink` should both inherit from a common (non-`env:`) block instead of keeping this current relationship.